File: src/playground/play.py
#playing with python constructs

#maps and lists (slices in Go)

class Student:
    def __init__(self, id, name):
        self.id = id
        self.name = name

def enrollStudents(course, students):
    for s in students:
        course[s.id] = s

def listStudents(course):
    for s in course:
        print course[s].id, course[s].name

def main():
    students = [Student(1,"alice"), Student(2, "bob"), Student(3,"Cathy")]
    coursePhysics = {}
    enrollStudents(coursePhysics, students)
    listStudents(coursePhysics)

if __name__ == "__main__":
    main()

File: src/playground/play.go
package main

import "fmt"

/*
 * temp file to play with various features of Go lang.
 */

//Student record
type Student struct {
	id   int    //id of the student.
	name string //name of the student.
}

func main() {
	//playing with slices and maps.
	var students = []Student{{1, "alice"}, {2, "bob"}, {3, "cathy"}}
	var sMap = make(map[int]Student)

	//adding new item to a slice. append doubles the capacity during each op
	students = append(students, Student{4, "deesha"})

	//range over slice returns the index and the object.
	for _, s := range students {
		sMap[s.id] = s
	}

	//range over maps returns the key and the value
	for k, v := range sMap {
		fmt.Println(k, v.name)
	}

	fmt.Printf("Total students = %d, capacity = %d\n", len(students), cap(students))
}
